tendrl/node_agent/provisioner/gluster/provisioner_base.py

```diff
@@ -0,0 +1,38 @@
+from abc import abstractmethod
+import six
+
+
+class PluginMount(type):
+
+    def __init__(cls, name, bases, attrs):
+        if not hasattr(cls, 'plugins'):
+            cls.plugins = []
+        else:
+            cls.register_plugin(cls)
+
+    def register_plugin(cls, plugin):
+        instance = plugin()
+        cls.plugins.append(instance)
+
+
+@six.add_metaclass(PluginMount)
+class ProvisionerBasePlugin(object):
+
+    @abstractmethod
+    def setup_gluster_node(self, hosts, packages, repo):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def create_gluster_cluster(self, hosts):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def gluster_provision_bricks(self, brick_dictionary, disk_type,
+                                 disk_count, stripe_count):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def gluster_volume_create(self, volume_name, brick_details,
```

Comment:
Do we plan to add more functions like delete_volume, set_volume_option,start/stop_volume etc later here?